### PR TITLE
WIP: Add chartvalues e2e test

### PR DESCRIPTION
**Pull request by indyfree**

### CODE REVIEW

File: integration/chart/chart.go [hunk 1]
```diff
@@ -53,11 +54,6 @@ func Push(h *framework.Host, charts []Chart) error {
 		return microerror.Mask(err)
 	}
 
-	l, err := micrologger.New(micrologger.Config{})
```
Comment (indyfree) [inline integration/chart/chart.go]: Boyscouting: Pass logger so we don't have to create an own here.

File: integration/test/chartvalues/chartvalues_test.go [hunk 2]
```diff
@@ -26,18 +34,36 @@ func TestChartValues(t *testing.T) {
...
 	}
-	err := chart.Push(h, charts)
+	err := chart.Push(l, h, charts)
 	if err != nil {
 		t.Fatalf("could not push inital charts to cnr %v", err)
 	}
 
-	// Test Creation
+	// Install configmap containing the values
+	err = helmClient.InstallFromTarball("/e2e/fixtures/tb-configmap-chart-1.0.0.tgz",
```
Comment (indyfree) [inline integration/test/chartvalues/chartvalues_test.go]: Install the values-configmap directly here.

File: integration/test/chartvalues/chartvalues_test.go [hunk 3]
```diff
@@ -47,4 +73,28 @@ func TestChartValues(t *testing.T) {
...
+
+	// Check if values are applied
+	rc, err := helmClient.GetReleaseContent(testChartRelease)
+	if err != nil {
+		t.Fatalf("could not get release content of %s %v", testChartRelease, err)
+	}
+	l.Log("level", "debug", "message", fmt.Sprintf("chart %s has values %#v", testChartRelease, rc.Values))
+
+	expectedValue := "values-config"
+	if rc.Values["config"] != expectedValue {
```
Comment (indyfree) [inline integration/test/chartvalues/chartvalues_test.go]: This is the actual test if chart-operator correctly applied the provided values (in the configmap) to the helm chart.

**Comment by indyfree** [comment]
@rossf7 If you wanna take over here, while I am on holidays its cool with me. Otherwise I can also pick it up afterwards. I added a more generic `tb-resource-chart` that should be easily extendable to secrets and user-configmap. Passing values with the configmap is working fine. 

There is still this issue https://github.com/giantswarm/giantswarm/issues/3896, but having a 10min wait for release status seems it to fix for now. (e2e-harness pr still needs to be merged aswell)

I didn't quite get the added secret support to work, and left it out for now. There are some troubles parsing the json string (maybe just need bas64 encoding in the chart, or also some adjustment in chart-operator itself, cc @MarcelMue ):
```
{rpc error: code = Unknown desc = release tb-configmap failed: Secret in version \"v1\" cannot be handled as a Secret: v1.Secret: ObjectMeta: v1.ObjectMeta: TypeMeta: Kind: Data: decode base64: illegal base64 data at input byte 0, error found in #10 byte of ...|secret\\\"}\"},\"kind\":\"|..., bigger context ...|\":{\"secret.json\":\"{\\\"secret\\\":\\\"default-secret\\\"}\"},\"kind\":\"Secret\",\"metadata\":{\"name\":\"values-secre|...}]
```

**Comment by rossf7** [comment]
Closing as this is stale. I'll update chart-operator e2e to use chart CRD as part of giantswarm/giantswarm#4949.

**Pull request closed without merging**